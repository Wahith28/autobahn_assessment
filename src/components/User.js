import { Typography } from "@mui/material";
import React from "react";
import "./user.css";

const User = ({ id, email, name, onDelete }) => {
  const handleDelete = () => {
    onDelete(id);
  };

  return (
    <tr testData="test">
      <td>
        <Typography>{name}</Typography>
      </td>
      <td>
        <Typography>{email}</Typography>
      </td>
      {/* <td>
                <button style={{margin:"5px"}}>edit</button>
                <button style={{margin:"5px"}} onClick={handleDelete}>delete</button>
            </td> */}
    </tr>
  );
};

export default User;
