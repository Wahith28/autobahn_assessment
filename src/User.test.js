import React from "react";
import { shallow } from "enzyme";
import User from "./components/User";
describe("User Component", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<User />);
    const a = component.find("[testData='test']").length;
    expect(a).toBe(1);
  });

  it("should render correctly", () => {
    const component = shallow(<User />);
    expect(component).toMatchSnapshot();
  });
});
