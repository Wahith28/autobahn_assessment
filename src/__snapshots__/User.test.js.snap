// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component should render correctly 1`] = `ShallowWrapper {}`;
